### DISCUSSION TITLE
Build from source error libQt5Core.so.5.15.2: undefined reference to

### DISCUSSION THREAD

**Discussion by SantiDiazC** [sofa-framework/sofa#5451] [Build / Config / Environment] [answered]
Hi, I tried to build SOFA from source following the instructions on the website and the youtube video (Linux compilation), 
however, it crashes at the end with the following error:

> [100%] Built target SceneChecking_test
>[100%] Linking CXX executable ../../../bin/runSofa
/usr/bin/ld: /home/hri/anaconda3/lib/libQt5Core.so.5.15.2: undefined reference to ''std::__exception_ptr::exception_ptr::_M_release()@CXXABI_1.3.13'
/usr/bin/ld: /home/hri/anaconda3/lib/libQt5Widgets.so.5.15.2: undefined reference to 'std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
/usr/bin/ld: /home/hri/anaconda3/lib/libQt5Core.so.5.15.2: undefined reference to `std::__exception_ptr::exception_ptr::_M_addref()@CXXABI_1.3.13'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [applications/projects/runSofa/CMakeFiles/runSofa.dir/build.make:159: bin/runSofa-22.12.00] Error 1
make[1]: *** [CMakeFiles/Makefile2:18991: applications/projects/runSofa/CMakeFiles/runSofa.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

![Selection_003](https://user-images.githubusercontent.com/9308405/235440349-dfdb7d21-c788-429f-8aa1-4fc125d6e676.png)

I added the CMAKE_PREFIX_PATH as suggested in the Troubleshooting CMake errors section, and the CMAKE configuration looks like the image attached.

Does anyone know how to solve this problem? Thank you in advance!

**Comment by fredroy** [comment, marked as answer]
The Qt you downloaded (and you set in CMAKE_PREFIX_PATH) is not the same as the ones detected by CMake.
CMake auto-detected the one you downloaded through anaconda through other packages.
So I suppose the compiler is using the headers of the Qt in /home/hri/Qt and the linker is using the libs in /home/hri/anaconda, hence the symbols error about glibc.

2 choices:
 - you only use the one installed by anaconda (and remove all reference from the one you installed)
 - you force cmake to detect the Qt you installed
